app/assets/javascripts/application.js

```diff
@@ -238,6 +238,19 @@ var Downloads = {
     Downloads.observeGUIOSFilter();
     Downloads.observePopState();
     Downloads.filterGUIS();
+    Downloads.autoDownload();
+  },
+
+  autoDownload: function() {
+    var ref = document.referrer;
+    var src = $('#auto-download-link').attr('href');
+
+    if (ref && src && ref.indexOf(document.location.origin) === 0) {
```

Comment:
nitpick: can you extract the if condition into a method with a clear name please? :)

Reply:
Maybe I'm missing your point.  But it already is in a method named `triggerAutoDownload`.  Well, technically, it was in a method I named `autoDownload` but then renamed it to the former.  If this is not what you meant, please elaborate.
```
  triggerAutoDownload: function() {
    var ref = document.referrer;
    var src = $('#auto-download-link').attr('href');

    if (ref && src && ref.indexOf(document.location.origin) === 0) {
      $('.downloading .hide').show();
      document.location = src;
    }
  },
```

Reply:
maybe I was not clear, sorry :) 

My suggestion was to move the `ref && src && ref.indexOf(document.location.origin) === 0` into a method (for instance `shouldAutoDownload(ref, src)`)

Reply:
I understand now.  Thanks for clarifying.  Change made.